plugins/nf-wave/src/main/io/seqera/wave/config/SpackOpts.java

```diff
@@ -27,32 +27,33 @@
  */
 public class SpackOpts {
 
-    final static public String DEFAULT_SPACK_BUILDER_IMAGE = "spack/ubuntu-jammy:v0.19.2";
-    final static public String DEFAULT_SPACK_RUNNER_IMAGE = "ubuntu:22.04";
-    final static public String DEFAULT_SPACK_OSPACKAGES = "";
-    final static public String DEFAULT_SPACK_FLAGS = "-O3";
-
-    public final Boolean checksum;
-    public final String builderImage;
-    public final String runnerImage;
-    public final String osPackages;
-    public final String cFlags;
-    public final String cxxFlags;
-    public final String fFlags;
+    /**
+     * The default base image from which the resulting container is created
+     */
+    final static public String DEFAULT_SPACK_BASE_IMAGE = "ubuntu:22.04";
```

Comment:
I have some thoughts here on the original builder vs runner image:
```
    final static public String DEFAULT_SPACK_BUILDER_IMAGE = "spack/ubuntu-jammy:v0.20.0";
    final static public String DEFAULT_SPACK_RUNNER_IMAGE = "ubuntu:22.04";
```

1. Note how the OS (type+version) should be consistent between the two. Hence, if the builder images cannot be chosen, the runner one should not be customisable neither (or viceversa, they should be flexible).

2. Depending on the model which is adopted for growing the binary cache, it might be good to have an option to choose the Spack version, i.e. the tag of the builder image. The scenario which I am thinking about is one where we advertise the ability for endusers to add new Spack recipes and use them promptly with Wave. In this case, the Spack version that needs to be used would not be the release one, but the next immediate weekly develop snapshot version, which will contain the newly added recipes. Or in alternative: we clearly state that only recipes available in the current release version of Spack are supported. 

I think this is an important point for the overall design of the service.

What do you think?

Reply:
Are you suggesting the runner image should be defined on backend side?

Reply:
Ok, i've moved it to the backend